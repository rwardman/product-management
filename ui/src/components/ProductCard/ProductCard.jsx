import React from "react";

import PropTypes from "prop-types";
import * as Styled from "./ProductCard.styles";

const ProductCard = ({
  id,
  name,
  description,
  category,
  creationDate,
  updateDate,
  lastPurchasedDate,
}) => {
  return (
    <Styled.Container>
      <Styled.TopContainer>
        <Styled.Title>{name}</Styled.Title>
        <div>
          <Styled.SubTitle> Id: </Styled.SubTitle>
          {id}
        </div>
      </Styled.TopContainer>

      <Styled.BottomContainer>
        <Styled.Paragraph>{description}</Styled.Paragraph>
        <Styled.InfoContainer>
          <Styled.SubTitle>Category: </Styled.SubTitle>
          {category}
        </Styled.InfoContainer>
        <Styled.InfoContainer>
          <Styled.SubTitle>Created: </Styled.SubTitle>
          {creationDate}
        </Styled.InfoContainer>
        <Styled.InfoContainer>
          <Styled.SubTitle>Updated: </Styled.SubTitle>
          {updateDate}
        </Styled.InfoContainer>
        <Styled.InfoContainer>
          <Styled.SubTitle>Last Purchased: </Styled.SubTitle>
          {lastPurchasedDate}
        </Styled.InfoContainer>
      </Styled.BottomContainer>
    </Styled.Container>
  );
};

export default ProductCard;

ProductCard.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string,
  description: PropTypes.string,
};
