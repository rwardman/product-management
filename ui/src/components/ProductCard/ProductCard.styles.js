import styled from "styled-components";

const Title = styled.span`
  font-weight: bold;
`;

const SubTitle = styled.span`
  font-weight: bold;
  font-size: 16px;
`;

const Container = styled.div`
  margin-right: auto;
  margin-left: auto;
  margin-top: 10px;
  width: 45%;
  min-width: 300px;
  max-width: 600px;
`;

const TopContainer = styled.div`
  display: flex;
  background-color: ${(props) => getProductCardColor(props.category)};
  justify-content: space-between;
  border-bottom: 1px solid #d8d8d8;
  padding: 10px;
`;

const BottomContainer = styled.div`
  display: flex;
  flex-direction: column;
  padding: 10px;
`;

const Paragraph = styled.div`
  padding-bottom: 10px;
`;

const InfoContainer = styled.div`
  padding-bottom: 5px;
`;

export {
  Title,
  SubTitle,
  Container,
  TopContainer,
  BottomContainer,
  Paragraph,
  InfoContainer,
};

const getProductCardColor = (category) => {
  switch (category) {
    case "Kitchen":
      return "#fffdd9";
    case "Power Tools":
      return "#ffe4d9";
    case "Furniture":
      return "#ffd9e9";
    case "Electric":
      return "#e3ffd9";
    case "Washroom":
      return "#d9e9ff";
    case "Textiles":
      return "#efffd9";
    case "Misc.":
      return "#e6d9ff";
    default:
      return "#eeedff";
  }
};
