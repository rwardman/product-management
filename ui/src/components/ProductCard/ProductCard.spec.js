import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import ProductCard from "./ProductCard";

test("The sort by component should be displayed correctly", () => {
  const tree = renderer
    .create(
      <ProductCard
        id="id"
        name="name"
        description="description"
        category="category"
        creationDate="creation date"
        updateDate="update date"
        lastPurchasedDate="last purchased date"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
