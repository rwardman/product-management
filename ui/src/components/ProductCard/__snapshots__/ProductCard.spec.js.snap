// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The sort by component should be displayed correctly 1`] = `
.c2 {
  font-weight: bold;
}

.c3 {
  font-weight: bold;
  font-size: 16px;
}

.c0 {
  margin-right: auto;
  margin-left: auto;
  margin-top: 10px;
  width: 45%;
  min-width: 300px;
  max-width: 600px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #eeedff;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: 1px solid #d8d8d8;
  padding: 10px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 10px;
}

.c5 {
  padding-bottom: 10px;
}

.c6 {
  padding-bottom: 5px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      name
    </span>
    <div>
      <span
        className="c3"
      >
         Id: 
      </span>
      id
    </div>
  </div>
  <div
    className="c4"
  >
    <div
      className="c5"
    >
      description
    </div>
    <div
      className="c6"
    >
      <span
        className="c3"
      >
        Category: 
      </span>
      category
    </div>
    <div
      className="c6"
    >
      <span
        className="c3"
      >
        Created: 
      </span>
      creation date
    </div>
    <div
      className="c6"
    >
      <span
        className="c3"
      >
        Updated: 
      </span>
      update date
    </div>
    <div
      className="c6"
    >
      <span
        className="c3"
      >
        Last Purchased: 
      </span>
      last purchased date
    </div>
  </div>
</div>
`;
